src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -100,10 +120,28 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
 
                 TypeSymbol declType = declarations[0].DeclaredType.Type;
 
+                MethodSymbol disposeMethod = TryFindDisposePattern(declType, diagnostics);
+
                 if (declType.IsDynamic())
                 {
                     iDisposableConversion = Conversion.ImplicitDynamic;
                 }
+                else if ((object)disposeMethod != null)
+                {
+                    BoundStatement boundBody2 = originalBinder.BindPossibleEmbeddedStatement(_syntax.Statement, diagnostics);
```

Comment:
This code, the code above, and the last return look identical. This could all probably be shared by moving the `disposeMethod` variable to the top of the method and flowing the variable to the final return.